ccxt.d.ts

```diff
@@ -175,11 +175,21 @@ declare module 'ccxt' {
         total: number;
     }
 
+    export interface BalanceCurrency {
+        currency: string;
+        price: number;
```

Comment:
It should be a different structure here:

- https://github.com/ccxt/ccxt/wiki/Manual#balance-structure (no such thing as `currency` and `price` inside).

Reply:
in the instructions you can see the structure under the example "free".
"free", "used" and "total" are a array with the pair currency and price.
in some cases it makes sense to access these arrays directly

Reply:
> in the instructions you can see the structure under the example "free".
"free", "used" and "total" are a array with the pair currency and price.

That contradicts to the Manual, the `free`, `used` and `total` are associative arrays that contain currency codes as keys and the balances as values. In other words, the edits in this PR do not align with the actual exchange implementations and with the docs.

Reply:
For example:

```
~/ccxt $ node examples/js/cli bitmart fetchBalance
2021-09-29T07:33:26.398Z
Node.js: v14.15.4
CCXT v1.57.13
bitmart.fetchBalance ()
{  info: { message:   "OK",
              code:   "1000",
             trace:   "72c8a9e2-a285-43d7-87eb-49ea65c0d1d5",
              data: { wallet: [ {        id: "BTC",
                                       name: "Bitcoin",
                                  available: "0.00003481",
                                     frozen: "0.00000000"  },
                                {        id: "ETH",
                                       name: "Ethereum",
                                  available: "0.00002277",
                                     frozen: "0.00000000"  },
                                {        id: "BMX",
                                       name: "BitMart Token",
                                  available: "993.95177422",
                                     frozen: "0.00000000"     },
                                {        id: "USDT",
                                       name: "Tether USD",
                                  available: "1947.49274039",
                                     frozen: "0.00000000"     }  ] } },
    BTC: { free: 0.00003481, used: 0, total: 0.00003481 },
    ETH: { free: 0.00002277, used: 0, total: 0.00002277 },
    BMX: {  free: 993.95177422,
            used: 0,
           total: 993.95177422  },
   USDT: {  free: 1947.49274039,
            used: 0,
           total: 1947.49274039  },
   free: {  BTC: 0.00003481,
            ETH: 0.00002277,
            BMX: 993.95177422,
           USDT: 1947.49274039 },
   used: { BTC: 0, ETH: 0, BMX: 0, USDT: 0 },
  total: {  BTC: 0.00003481,
            ETH: 0.00002277,
            BMX: 993.95177422,
           USDT: 1947.49274039 }                                        }
~/ccxt $
```

Reply:
hmm, i do not understand that... :-(
is it a naming problem?
i can also rename the fields.

because of the addition of **free**, **used** and **total** it shouldn't really be because they were newly added.

The current problem is that TS users cannot use **free**, **used** and **total** at the moment.

Reply:
@cabanossi take a look at the above structure – the values under `free`, `used` and `total` are not arrays, they are associative objects. They don't have any keys like `currency` and `price` – do you see that?

Reply:
@cabanossi check this commit: https://github.com/ccxt/ccxt/pull/10079/commits/76d5e8e333973c3c3e833690cdea1dd5c8285895

Reply:
sorry, i see it now!!!
thanks a lot for correction!!!

next time i'll be more careful!

Reply:
@cabanossi thank you for your involvement!